expression/builtin_math.go

```diff
@@ -843,38 +844,36 @@ func (b *builtinAtanSig) eval(row []types.Datum) (d types.Datum, err error) {
 	return d, nil
 }
 
+//cos start
 type cosFunctionClass struct {
 	baseFunctionClass
 }
 
 func (c *cosFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinCosSig{newBaseBuiltinFunc(args, ctx)}
+	tp := types.NewFieldType(mysql.TypeDouble)
+	tp.Flen = 8
```

Comment:
According to [MySQL source code](https://github.com/mysql/mysql-server/blob/5.7/sql/item_func.cc#L2813), we may set it to 23 (DBL_DIG + 8)